deepchem/models/torch_models/cgcnn.py

```diff
@@ -0,0 +1,255 @@
+  by nodes representing atom properties and edges representing connections between atoms
+  in a crystal. And, this model updates the node representations using both neighbor node
+  and edge representations. Please confirm the detail algorithms from [1]_.
+
+  References
+  ----------
+  .. [1] Xie, Tian, and Jeffrey C. Grossman. "Crystal graph convolutional neural networks
+     for an accurate and interpretable prediction of material properties." Physical review letters
+     120.14 (2018): 145301.
+
+  Notes
+  -----
+  This class requires DGL and PyTorch to be installed.
+  """
+
+  def __init__(
+      self,
+      in_node_dim: int = 92,
+      hidden_node_dim: int = 64,
+      in_edge_dim: int = 41,
+      num_conv: int = 3,
+      predicator_hidden_feats: int = 128,
+      n_out: int = 1,
+  ):
+    """
+    Parameters
+    ----------
+    in_node_dim: int, default 92
+      The length of the initial node feature vectors. The 92 is
+      based on length of vectors in the atom_init.json.
+    hidden_node_dim: int, default 64
+      The length of the hidden node feature vectors.
+    in_edge_dim: int, default 41
+      The length of the initial edge feature vectors. The 41 is
+      based on default setting of CGCNNFeaturizer.
+    num_conv: int, default 3
+      The number of convolutional layers.
+    predicator_hidden_feats: int, default 128
+      Size of hidden graph representations in the predicator, default to 128.
+    n_out: int
+      Number of the output size, default to 1.
+    """
+    super(CGCNN, self).__init__()
+    self.embedding = nn.Linear(in_node_dim, hidden_node_dim)
+    self.convs = [
+        CGCNNLayer(
+            hidden_node_dim=hidden_node_dim,
+            edge_dim=in_edge_dim,
+            batch_norm=True) for _ in range(num_conv)
+    ]
+    self.fc = nn.Linear(hidden_node_dim, predicator_hidden_feats)
+    self.out = nn.Linear(predicator_hidden_feats, n_out)
+
+  def forward(self, dgl_graph):
+    """Predict labels
+
+    Parameters
+    ----------
+    dgl_graph: DGLGraph
+      DGLGraph for a batch of graphs. The graph expects that the node features
+      are stored in `ndata['x']`, and the edge features are stored in `edata['edge_attr']`.
+
+    Returns
+    -------
+    out: torch.Tensor
+      The output value, the shape is `(batch_size, n_out)`.
+    """
+    try:
+      import dgl
+    except:
+      raise ValueError("This class requires DGL to be installed.")
+
+    graph = dgl_graph
+    # embedding node features
+    graph.ndata['x'] = self.embedding(graph.ndata['x'])
+
+    # convolutional layer
+    for conv in self.convs:
+      graph = conv(graph)
+
+    # pooling
+    graph_feat = dgl.mean_nodes(graph, 'x')
+    graph_feat = self.fc(graph_feat)
+    out = self.out(graph_feat)
+    return out
+
+
+class CGCNNModel(TorchModel):
+  """CGCNN wrapper model for converting PyTorch style to Keras style.
+
+  Please confirm the details about CGCNN from CGCNN class docstring.
```

Comment:
For most users, this is the class they'll be working with.  This one is the DeepChem model, and the other classes are internal implementation details that they won't look at unless they're doing something advanced.  So this class is where the main documentation should go.

Reply:
OK!.

Reply:
It still uses a description that will be confusing for users: "CGCNN wrapper model for converting PyTorch style to Keras style."  Just copy over the docstring exactly (except using the appropriate example code):

```
"""Crystal Graph Convolutional Neural Network (CGCNN).
This model takes arbitary crystal structures as an input, and predict material properties
using the element information and connection of atoms in the crystal. If you want to get
...
```

That way it just tells the user what this class is (a CGCNN) and tells them how to use it, but doesn't confuse them with implementation details that aren't relevant to them.

Reply:
I changed the correct statement!